typings/styled-components.d.ts

```diff
@@ -1,163 +1,176 @@
-  image: StyledFunction<React.ComponentClass<React.SVGProps>, React.SVGAttributes<SVGImageElement>>;
-  line: StyledFunction<React.ComponentClass<React.SVGProps>, React.SVGAttributes<SVGLineElement>>;
-  linearGradient: StyledFunction<React.ComponentClass<React.SVGProps>, React.SVGAttributes<SVGLinearGradientElement>>;
-  mask: StyledFunction<React.ComponentClass<React.SVGProps>, React.SVGAttributes<SVGMaskElement>>;
-  path: StyledFunction<React.ComponentClass<React.SVGProps>, React.SVGAttributes<SVGPathElement>>;
-  pattern: StyledFunction<React.ComponentClass<React.SVGProps>, React.SVGAttributes<SVGPatternElement>>;
-  polygon: StyledFunction<React.ComponentClass<React.SVGProps>, React.SVGAttributes<SVGPolygonElement>>;
-  polyline: StyledFunction<React.ComponentClass<React.SVGProps>, React.SVGAttributes<SVGPolylineElement>>;
-  radialGradient: StyledFunction<React.ComponentClass<React.SVGProps>, React.SVGAttributes<SVGRadialGradientElement>>;
-  rect: StyledFunction<React.ComponentClass<React.SVGProps>, React.SVGAttributes<SVGRect>>;
-  stop: StyledFunction<React.ComponentClass<React.SVGProps>, React.SVGAttributes<SVGStopElement>>;
-  svg: StyledFunction<React.ComponentClass<React.SVGProps>, React.SVGAttributes<SVGSVGElement>>;
-  text: StyledFunction<React.ComponentClass<React.SVGProps>, React.SVGAttributes<SVGTextElement>>;
-  tspan: StyledFunction<React.ComponentClass<React.SVGProps>, React.SVGAttributes<SVGTSpanElement>>;
+  circle: SvgStyledFunction<SVGCircleElement>;
+  clipPath: SvgStyledFunction<SVGClipPathElement>;
+  defs: SvgStyledFunction<SVGDefsElement>;
+  ellipse: SvgStyledFunction<SVGEllipseElement>;
+  g: SvgStyledFunction<SVGGElement>;
+  image: SvgStyledFunction<SVGImageElement>;
+  line: SvgStyledFunction<SVGLineElement>;
+  linearGradient: SvgStyledFunction<SVGLinearGradientElement>;
+  mask: SvgStyledFunction<SVGMaskElement>;
+  path: SvgStyledFunction<SVGPathElement>;
+  pattern: SvgStyledFunction<SVGPatternElement>;
+  polygon: SvgStyledFunction<SVGPolygonElement>;
+  polyline: SvgStyledFunction<SVGPolylineElement>;
+  radialGradient: SvgStyledFunction<SVGRadialGradientElement>;
+  rect: SvgStyledFunction<SVGRectElement>;
+  stop: SvgStyledFunction<SVGStopElement>;
+  svg: SvgStyledFunction<SVGSVGElement>;
+  text: SvgStyledFunction<SVGTextElement>;
+  tspan: SvgStyledFunction<SVGTSpanElement>;
 }
 
 declare const styled: StyledInterface;
 
-export const css: StyledFunction<(string | Function)[], any>;
-export const keyframes: StyledFunction<string, any>;
-export const injectGlobal: StyledFunction<undefined, any>;
+export function css<P>(strings: TemplateStringsArray, ...interpolations: Interpolation<StyledProps<P>>[]): Interpolation<StyledProps<P>>[];
+export function keyframes(strings: TemplateStringsArray, ...interpolations: (string | number)[]): string;
+export function injectGlobal(strings: TemplateStringsArray, ...interpolations: (string | number)[]): void;
```

Comment:
In fact, functions are disallowed here. Since it is a 'global' context of execution.
I checked this with code and they're just stringify interpolations with invoking any functions. That means they need to be disallowed.